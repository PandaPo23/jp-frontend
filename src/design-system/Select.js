import styled from '@emotion/styled';
import React from 'react';
import { themeGet } from 'styled-system';

import system from '../utils/System';
import Flex from './Flex';
import Icon from './Icon';

const ClickableIcon = styled(Icon)`
  pointer-events: none;
`;

const SelectBase = system(
  'SelectBase',
  {
    as: 'select',
    fontSize: 1,
    m: 0,
    borderRadius: 'sm',
    border: 1,
    p: 2,
    bg: 'transparent',
  },
  ({ theme }) => ({
    appearance: 'none',
    cursor: 'pointer',
    marginRight: `calc(-${theme.space[2]} - 24px)`,
    paddingRight: `calc(${theme.space[2]} + ${theme.space[1]} + 24px)`,
    '&:hover': {
      borderColor: themeGet('colors.blue'),
      boxShadow: `0 0 0 1px ${themeGet('colors.blue')}`,
    },
  })
);

export const Select = styled((props) => (
  <Flex alignItems="center" justifyItems="center">
    <SelectBase {...props} />
    <ClickableIcon ml={1} name="arrow-drop-down" color="grays.5" />
  </Flex>
))``;

export default Select;
