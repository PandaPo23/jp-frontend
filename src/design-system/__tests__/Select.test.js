import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Select from '../Select';

describe('Select component', () => {
  const getComponent = (props) => (
    <Select>
      <option value="1">1</option>
      <option value="2">2</option>
    </Select>
  );
  it('should render as Select component', () => {
    const wrapper = shallow(<Select />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render a select component with 2 options', () => {
    const component = getComponent();
    expect(component.props.children.length).toBe(2);
  });
});
