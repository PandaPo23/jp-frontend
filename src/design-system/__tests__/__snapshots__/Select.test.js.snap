// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render as Select component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
